src/diagnostic-channel-publishers/src/pg.pub.ts

```diff
@@ -0,0 +1,164 @@
+                // since we are always inserting ourselves as a callback, we have to restore the original
+                // behavior if the user didn't provide one themselves
+                if (err) {
+                    if (cb) {
+                        return cb.apply(this, arguments);
+                    } else if (queryResult && queryResult instanceof EventEmitter) {
+                        queryResult.emit("error", err);
+                    }
+                } else if (cb) {
+                    cb.apply(this, arguments);
+                }
+            });
+
+            if (cachePatched) {
+                // this should only happen in the case of config.callback = patch(config.callback)
+                patchedCallbacks.set(trackingCallback, trackingCallback);
+            }
+
+            if (cb) {
+                patchedCallbacks.set(cb, trackingCallback);
+            }
+
+            return trackingCallback;
+        }
+
+        // this function takes too many variations of aguments.
+        // this patches any provided callback or creates a new callback if one wasn't provided.
+        // since the callback is always called (if provided) in addition to always having a Promisified
+        // EventEmitter returned (well, sometimes -- see above), its safe to insert a callback if none was given
+        if (typeof config === "string") {
+            if (values instanceof Array) {
+                data.query.preparable = {
+                    text: config,
+                    args: values,
+                };
+                callback = patchCallback(callback);
+            } else {
+                data.query.text = config;
+
+                // pg v6 will, for some reason, accept both
+                // client.query("...", undefined, () => {...})
+                // **and**
+                // client.query("...", () => {...});
+                // Internally, precedence is given to the callback argument
+                if (callback) {
+                    callback = patchCallback(callback);
+                } else {
+                    values = patchCallback(values);
+                }
+            }
+        } else {
+            if (typeof config.name === "string") {
+                data.query.plan = config.name;
+            } else if (config.values instanceof Array) {
+                data.query.preparable = {
+                    text: config.text,
+                    args: config.values,
+                };
+            } else {
+                data.query.text = config.text;
+            }
+
+            if (callback) {
+                callback = patchCallback(callback);
+            } else if (values) {
+                values = patchCallback(values);
+            } else {
+                config.callback = patchCallback(config.callback, true);
+            }
+        }
+
+        queryResult = originalClientQuery.call(this, config, values, callback);
```

Comment:
this should always call `originalClient.apply(this, arguments)`, as we want to preserve whatever the user is doing, right or wrong.  What's happening here is, if you user  is calling this:

```
   pg.query(myConfig, myValues, myCallback, 1, 2, 3, 4, 5, 6, 7, 8); 
```

You're stripping off the last 8 arguments.  Now, it should matter, but in principle our proxies shouldn't have side affects like this. 

Also note that if call `(originalQuery.apply(this, arguments)`, you need to make sure you update the arguments array with whatever you've replaced.